addons/io_scene_gltf2/blender/exp/gltf2_blender_gather.py

```diff
@@ -154,6 +152,21 @@ def __gather_animations(blender_scene, export_settings):
 
     return new_animations
 
+def __gather_materials(blender_scene, export_settings):
+    materials = []
+
+    for _blender_object in blender_scene.objects:
+        blender_object = _blender_object.proxy if _blender_object.proxy else _blender_object
+        for slot in blender_object.material_slots:
+            # Check if this slot as assigned material
+            if slot.material is None:
```

Comment:
@scurest For than i understand, you thing to change `if slot.material is None:` to `if slot.material is None or slot.material == blender_object.active_material:` ?

For exclude active_material in __gather_materials array for better performance ?

Reply:
Er, no. I was thinking instead of looking in any material slot in *any object in an exported scene*, only looking in any material slot in *an object that gets exported*.

Is that what the lastest "Check if this object is exclude from view layer" check is supposed to do?

Reply:
You think is better to use `bpy.data.objects` directly ?
But i need `blender_scene` with `gltf2_blender_gather_nodes.gather_node`, i can use `users_scene` with Object but it's probably weird code because `users_scene` is tuple.

"Check if this object is exclude from view layer" is because if you uncheck collection, all objects associated with this collection are not exported and is not necessary to exported materials assigned to these objects

Reply:
I was thinking something like [this](https://github.com/KhronosGroup/glTF-Blender-IO/compare/9057985...scurest:unused-materials).

Reply:
Yes, your codes is more efficient.
But probably change setting name like "Variants materials" for `KHR_materials_variants` ?